src/plugins/mission/mission_impl.cpp

```diff
@@ -1024,6 +1024,49 @@ Mission::Result MissionImpl::build_mission_items(
     return result;
 }
 
+Mission::Result MissionImpl::import_simple_mission_item(
+    std::vector<Mission::MissionItem>& all_mission_items,
+    const Json::Value& json_mission_item,
+    MissionItem& new_mission_item)
+{
+    // Parameters of Mission item & MAV command of it.
+    MAV_CMD command = static_cast<MAV_CMD>(json_mission_item["command"].asInt());
+
+    // Extract parameters of each mission item
+    std::vector<double> params;
+    for (auto& p : json_mission_item["params"]) {
+        if (p.type() == Json::nullValue) {
+            // QGC sets params as `null` if they should be unchanged.
+            params.push_back(double(NAN));
+        } else {
+            params.push_back(p.asDouble());
+        }
+    }
+    return build_mission_items(command, params, new_mission_item, all_mission_items);
+}
+
+Mission::Result MissionImpl::import_complex_mission_item(
+    std::vector<Mission::MissionItem>& all_mission_items,
+    const Json::Value& json_complex_mission_item,
+    MissionItem& new_mission_item)
+{
+    if (json_complex_mission_item["TransectStyleComplexItem"].isNull()) {
+        LogWarn() << "Unknown complex item type (" << json_complex_mission_item["complexItemType"]
+                  << ")";
+        return Mission::Result::UnsupportedMissionCmd;
+    }
+
+    Json::Value complex_item = json_complex_mission_item["TransectStyleComplexItem"];
```

Comment:
Could you add a code comment for the reader on what this complex "TransectStyleComplexItem" actually is?

Reply:
I added a comment and a test case.
The tests output lots of warnings (unsupported mission items) for 206 (MAV_CMD_DO_SET_CAM_TRIGG_DIST) and 20 (MAV_CMD_NAV_RETURN_TO_LAUNCH ) which are used in the test plan file.

Reply:
Ok thanks. I guess that's expected because they are not understood by the simple MAVSDK mission API.